flink-java/src/main/java/org/apache/flink/api/java/functions/SemanticPropUtil.java

```diff
@@ -309,15 +308,20 @@ public static DualInputSemanticProperties getSemanticPropsDual(
 			getSemanticPropsDualFromString(result, forwardedFirst, forwardedSecond,
 					nonForwardedFirst, nonForwardedSecond, readFirst, readSecond, inType1, inType2, outType);
 			return result;
-		} else {
-			return new DualInputSemanticProperties();
 		}
+		return null;
+	}
+
+	public static void getSemanticPropsSingleFromString(SingleInputSemanticProperties result,
+														String[] forwarded, String[] nonForwarded, String[] readSet,
+														TypeInformation<?> inType, TypeInformation<?> outType) {
+		getSemanticPropsSingleFromString(result, forwarded, nonForwarded, readSet, inType, outType, false);
 	}
 
 	public static void getSemanticPropsSingleFromString(SingleInputSemanticProperties result,
 			String[] forwarded, String[] nonForwarded, String[] readSet,
-			TypeInformation<?> inType, TypeInformation<?> outType)
-	{
+			TypeInformation<?> inType, TypeInformation<?> outType,
+			boolean skipIncompatibleTypes) {
```

Comment:
Sometimes the analyzer works better than required. E.g. the analyzer outputs @ForwardedFields("*->record.customer.name") but if customer is a GenericType output type, the types are incompatible. I thought it is better to reuse the type compatibility checking of the PropUtil than reimplement everything, but skip types that are incompatible without throwing Exceptions.